hera_pspec/pspecdata.py

```diff
@@ -1621,9 +1698,14 @@ def get_Q_alt(self, mode, allow_fft=True):
         if mode >= self.spw_Ndlys:
             raise IndexError("Cannot compute Q matrix for a mode outside"
                              "of allowed range of delay modes.")
-
-        if (self.spw_Ndlys == self.spw_Nfreqs) and (allow_fft == True):
-            _m = np.zeros((self.spw_Nfreqs,), dtype=np.complex)
+        nfreq = self.spw_Nfreqs
+        if include_extension:
+            nfreq = nfreq + np.sum(self.filter_extension)
+            phase_correction = self.filter_extension[0]
+        else:
+            phase_correction = 0.
+        if (self.spw_Ndlys == nfreq) and (allow_fft == True):
```

Comment:
whether this evaluates to true will depend on if  include_extension is set to True . If include_extension is True, then we can only use fft mode if Ndlys = (spw_nfreq + 2 * n_ext). We typically want N_dlys to equal the number of channels in spw_nfreq since this is the number of frequencies being DFTd to obtain a bandpower estimate so I don't think set_Ndlys needs to be changed.

That said,  we sometimes want to call Q with Ndlys = Nfreq_spw but also for Q to include Nfreq = Next + Nfreq_spw (When computing dC / dp since C includes the channels in the filter extensions) which does not modify Ndlys but does change the shape of the Q matrix.